src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -848,6 +848,30 @@ public String getGitExe(Node builtOn, EnvVars env, TaskListener listener) {
         return null;
     }
 
+    String readGitDescribe(AbstractBuild<?, ?> ctx, TaskListener listener) {
+
+        BuildData buildData = ctx.getAction(BuildData.class);
+        if (buildData==null){
+            return "";
+        }
+        Revision rev = buildData.getLastBuiltRevision();
+        if (rev == null) {
+            return "";
+        }
+
+        try {
+            GitClient client = createClient(listener, ctx.getEnvironment(listener), ctx, ctx.getWorkspace());
+            return client.describe(rev.getSha1String());
+        } catch (GitException e) {
+            //ignore
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
```

Comment:
Implicit throwing of runtime exceptions is a bad practice. Consider throwing a common exception.
<code>InterruptedException</code>s should be also propagated as-is and not suppressed, because they are being used to terminate builds